### PR TITLE
feat(boards): #MAG-18 add boards (pre)deletion and restore

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- add toaster activating on board select
- add board delete, restore options to toaster
- add delete, restore boards API routes

## Checklist tests

- check if toaster is activating on board select
- check if delete is working properly : on "my boards" it should move the board to trash, on trash view it should delete the board permanently
- check if a restored board from trash view is moved back to the "my boards" view

## Issue ticket number and link

https://jira.support-ent.fr/browse/MAG-18

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/jsonschema/deleteBoard.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "boardIds": {
+      "type": [
+        "array",
+        "string"
+      ]
+    }
+  },
```
Comment (BastienBenard) [inline src/main/resources/jsonschema/deleteBoard.json:12]:
```suggestion
  "properties": {
    "boardIds": {
      "type": "array",
      "items": {
        "anyOf": [
          {
            "type": "string"
          }
        ]
      }
    },
```

File: src/test/java/fr/cgi/magneto/service/BoardServiceTest.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public void testGetAllBoardsQuery(TestContext ctx) throws Exception {
         testUser.setUserId("ownerId");
 
         JsonObject query = Whitebox.invokeMethod(this.boardService, "getAllBoardsQuery", testUser, 0, "test", "folderId",
-                false, true, "name", true);
+                false, true, false, "name", true);
```
Comment (BastienBenard) [inline src/test/java/fr/cgi/magneto/service/BoardServiceTest.java:79]: A voir, mais ce n'est pas suffisent pour vérifier que tes changements sont correcte. 
Ici tu test bien la non régression de ton code, mais pas la nouvelle fonctionnalité. Il faudrait faire un autre test avec isDeleted a true.

File: src/main/resources/public/ts/directives/board-delete-lightbox/board-delete-lightbox.directive.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    submitDeleteBoard = async (): Promise<void> => {
+        try {
+            if (this.isPredelete) {
+                this.boardsService.preDeleteBoards(this.boardIds)
+                    .then((response: AxiosResponse) => {
+                        if (response.status === 200 || response.status === 201) {
+                            toasts.confirm((this.boardIds.length > 1) ?
+                            'magneto.predelete.boards.confirm' : 'magneto.predelete.board.confirm');
+                            this.$scope.$parent.$eval(this.$scope['vm']['onSubmit'])();
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/board-delete-lightbox/board-delete-lightbox.directive.ts:48]: fais en $parse si tu peux (et typage de tes props)

File: src/main/resources/public/ts/directives/board-delete-lightbox/board-delete-lightbox.directive.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                        }
+                    }
+                );
+            } else {
+                this.boardsService.deleteBoards(this.boardIds)
+                    .then((response: AxiosResponse) => {
+                       if (response.status === 200 || response.status === 201) {
+                           toasts.confirm((this.boardIds.length > 1) ?
+                           'magneto.delete.boards.confirm' : 'magneto.delete.board.confirm');
+                           this.$scope.$parent.$eval(this.$scope['vm']['onSubmit'])();
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/board-delete-lightbox/board-delete-lightbox.directive.ts:63]: fais en $parse si tu peux (et typage de tes props)

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**